### DISCUSSION TITLE
preset for renovating renovate configuration itself

### DISCUSSION THREAD

**Discussion by electriquo** [renovatebot/renovate#23924] [Suggest an Idea]
### Tell us more.

when using [sharable configuration preset](https://docs.renovatebot.com/config-presets/), renovate configuration may look like

```json
{
  "extends": ["github>rarkins/renovate-config#1.2.3"]
}
```

and would expect renovate to renovate its configuration and open a pull request when there is a new version to to a sharable configuration preset. for instance, updating 1.2.3 to 1.2.4.

feels natural to add a preset to the [regex manager preset](https://docs.renovatebot.com/presets-regexManagers/) for updating renovate configuration itself.

what do you think?

**Comment by rarkins** [comment]
Sounds good

**Reply by electriquo** [reply]
maybe this can help?

```json
  "regexManagers": [
    {
      "fileMatch": ["(^|/).?renovate(?:rc)?(?:\\.json5?)?$"],
      "matchStrings": [
        "(.*?[^\"])\"(?<provider>git(lab|hub|ea))>(?<depName>.*?)((?<path>(\\/\\/)(.*?[^#]))?#(?<currentValue>.*?[^\"]))?\""
      ],
      "datasourceTemplate": "{{provider}}-tags"
    },
```

**Comment by jamietanna** [comment]
This is now available

**Reply by electriquo** [reply]
Please share the corresponding PR and any reference to the docs (E.g. how to enable this? is this the default behavior? which preset includes this?).
Thank you :)

**Reply by RahulGautamSingh** [reply]
It's not a preset. It's a built-in manager called `renovate-config`, enabled by default like any other manager. 

No preset to extend, nothing to turn on.

https://docs.renovatebot.com/modules/manager/renovate-config/
